### DISCUSSION TITLE
question about recieving

### DISCUSSION THREAD

**Discussion by Mer-Mer4** [bbernhard/signal-cli-rest-api#252] [Q&A]
I am working on a bot that works with X number of groups which means it needs to be able to just listen of incoming messages. From what i understand i need to call receive for each group every time i want an update, even with the json-rcp. is that accurate? id prefer to not use the pure Signal-Cli because dbus has been an absolute chore to get working in my container.

**Comment by bbernhard** [comment]
In `json-rpc` mode you will get notified when a new message arrives. While in `normal`/`native` mode the `receive` endpoint is a GET request, the `receive` endpoint in `json-rpc` mode is a websocket connection. See [https://github.com/bbernhard/signal-cli-rest-api/discussions/160] for details
